plugins/outputs/sematext/serializer/serializer.go

```diff
@@ -156,7 +166,98 @@ func serializeMetric(key string, value interface{}) string {
 // as possible in a single output line, based on tags and timestamp of each metric. When multiple metrics share the
 // same tags and the same timestamp, we can write them in a single line to reduce the total bulk size by not
 // repeating the same tags+timestamp multiple times.
-// TODO to be implemented in the future to make Telegraf requests to Sematext backend smaller
 type CompactMetricSerializer struct {
-	tagsIDToMetrics map[string]telegraf.Metric
+	log    telegraf.Logger
+	config *serializationConfig
+}
+
+// NewCompactMetricSerializer creates an instance of CompactMetricSerializer
+func NewCompactMetricSerializer(log telegraf.Logger) *CompactMetricSerializer {
+	return &CompactMetricSerializer{
+		log:    log,
+		config: NewSerializationConfig(),
+	}
+}
+
+// Write serializes input metrics array according to Sematext variant of influx line protocol. The output is returned
+// as []byte which can be empty if there were no metrics or metrics couldn't be serialized.
+func (s *CompactMetricSerializer) Write(metrics []telegraf.Metric) []byte {
+	var output bytes.Buffer
+	idToMetrics := make(map[string][]telegraf.Metric)
+
+	// first group the metrics that share the same identification
+	for _, m := range metrics {
+		id := buildId(m)
+		idToMetrics[id] = append(idToMetrics[id], m)
+	}
+
+	// sort the keys keep the order fixed
+	sortedIds := make([]string, 0, len(idToMetrics))
+	for i := range idToMetrics {
+		sortedIds = append(sortedIds, i)
+	}
+	sort.Strings(sortedIds)
+
+	// then create 1 metrics line for each of created groups
+	for _, groupId := range sortedIds {
```

Comment:
groupId -> groupID

Reply:
Yeah, forgot to run `make lint` before sending PR...